docs/topics/classes.md

```diff
@@ -1,245 +1,747 @@
+        // I am Alice, 20 years old, and I study at Engineering University.
     }
 }
+
+fun main() {
+    // Creates an object of the Student class
+    val student = Student("Alice", 20, "Engineering University")
+    // Calls the non-abstract member
+    student.greet()
+    // Calls the overridden abstract member
+    student.introduce()
+}
 ```
+{kotlin-runnable="true"}
+
+You don't need to annotate abstract classes or functions with the `open` keyword because they are implicitly 
+open and designed to be inherited. 
+
+Similarly, you can override a non-abstract `open` member with an abstract one:
 
-You can override a non-abstract `open` member with an abstract one.
+## Open keyword
+
+In Kotlin, the `open` keyword indicates that a class or a member (function or property) can be overridden in subclasses. 
+By default, Kotlin classes and their members are _final_, meaning they cannot be inherited or overridden unless explicitly marked as `open`:
 
 ```kotlin
-open class Polygon {
-    open fun draw() {
-        // some default polygon drawing method
+// Base class with the 'open' keyword to allow inheritance
+open class Person(
+    val name: String
+) {
+    // Open function that can be overridden in a subclass
+    open fun introduce() {
+        println("Hello, my name is $name.")
     }
 }
 
-abstract class WildShape : Polygon() {
-    // Classes that inherit WildShape need to provide their own
-    // draw method instead of using the default on Polygon
-    abstract override fun draw()
+// Subclass inheriting from 'Person' and overriding the 'introduce()' function
+class Student(
+    name: String,
+    val school: String
+) : Person(name) {
+    override fun introduce() {
+        println("Hi, I'm $name, and I study at $school.")
+    }
 }
 ```
 
+In addition, you need to add the `open` keyword to every property
+or function that can be overridden.
+
+If you override a member of a base class, the overriding member
+will also be open by default. If you want to change this and forbid the subclasses of your
+class from overriding your implementation, you can explicitly mark the overriding
+member as `final`:
+
+```kotlin
+// Base class with the 'open' keyword to allow inheritance
+open class Person(val name: String) {
+    // Open function that can be overridden in a subclass
+    open fun introduce() {
+        println("Hello, my name is $name.")
+    }
+}
+
+// Subclass inheriting from 'Person' and overriding the 'introduce()' function
+class Student(name: String, val school: String) : Person(name) {
+    // The 'final' keyword prevents further overriding in subclasses
+    final override fun introduce() {
+        println("Hi, I'm $name, and I study at $school.")
+    }
+}
+```
+
+You don't need to annotate abstract classes or functions with the `open` keyword because they are implicitly 
+open and designed to be inherited. 
+
+Similarly, you can override a non-abstract `open` member with an abstract one:
```

Comment:
Tbh, I do not think this example adds anything useful for one reading the documentation. Such pattern of overriding `open` with `abstract` is very strange and should not be used unless there are very specific circumstances.